### ISSUE TITLE
🐛 Markdown parser failures in our website repo

### ISSUE THREAD

**Issue by dyc3** [biomejs/biome#9727]
### Environment information

<details>

```bash

main, with markdown support enabled

```

</details>

### What happened?

Was doing some manual testing of the markdown parser by running it on all the files in our website repo.

These files fail.
- https://github.com/biomejs/website/blob/4b2f104a75fd3d86ddfc2354d8988e83296d5f8d/src/content/docs/ja/formatter/option-philosophy.md?plain=1#L6 fails with a panic (char boundary issue)
- https://github.com/biomejs/website/blob/4b2f104a75fd3d86ddfc2354d8988e83296d5f8d/src/content/docs/zh-CN/formatter/option-philosophy.md fails also fails with a panic (char boundary issue)
- https://github.com/biomejs/website/blob/4b2f104a75fd3d86ddfc2354d8988e83296d5f8d/src/content/docs/internals/changelog_v1.md emits a parsing diagnostic

### Expected result

Parsing to succeed

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by dyc3** [comment]
cc @jfmcdowell

**Comment by jfmcdowell** [comment]
@dyc3 -- ACK. Please assign to me

**Comment by jfmcdowell** [comment]
@dyc3 are you ok to close this?

**Comment by dyc3** [comment]
just tested the files again, didn't get any panicks. looks fixed to me
